src/components/ha-sidebar.ts

```diff
@@ -370,7 +370,9 @@ class HaSidebar extends LitElement {
   private _renderPanels(panels: PanelInfo[]) {
     return panels.map((panel) =>
       this._renderPanel(
-        panel.url_path,
+        panel.url_path === "hassio"
```

Comment:
Instead of doing this, we should let the `hassio` integration register the config panel with a config to highlight.

So replace https://github.com/home-assistant/core/blob/dev/homeassistant/components/hassio/__init__.py#L461

With https://github.com/home-assistant/core/blob/dev/homeassistant/components/config/__init__.py#L38-L40

But pass the focusedPath as a config: https://github.com/home-assistant/core/blob/dev/homeassistant/components/frontend/__init__.py#L267

Reply:
Ehhh... no?
Wouldn't that stop the panel from loading at all?
How does it know what element to load? and which file to load it from?

Reply:
Actually. We should still register the hassio custom panel but not register it as a sidebar thing

Reply:
The built-in panel should be registered for sidebar, although now I don't know if the frontend is able to sort it correctly HMMMMM

Reply:
Maybe your solution is the best? It's temporary anyway.

Reply:
I tested with regular panel, and every way I did it would result in a blank page.
Custom panels have the option for custom config as well, so I  can move that handling to the integration, but as you say this is temporary for a transition period only :man_shrugging:

Reply:
Even when we remove it, it would still need to be registered as a custom panel so we need to then have this element ignore `panel.url_path === "hassio"` instead

Reply:
You can register panels without a sidebar entry.